api/include/opentelemetry/logs/logger.h

```diff
@@ -55,65 +54,321 @@ class Logger
    * @throws No exceptions under any circumstances.
    */
 
-  /* The below method is a logging statement that takes in a LogRecord.
-   * A default LogRecord that will be assigned if no parameters are passed to Logger's .log() method
-   * which should at minimum assign the trace_id, span_id, and timestamp
+  /**
+   * Logs a LogRecord, which contains all the fields of the Log Data Model. Normally called
+   * indirectly from other Log() methods, but can be called directly for complex logs that have
+   * many fields and datatypes.
+   * @param record A log record filled with information from the user.
    */
-  virtual void log(const LogRecord &record) noexcept = 0;
+  virtual void Log(const LogRecord &record) noexcept = 0;
+
+  /// Overloaded Log methods, which are simpler than calling a LogRecord directly
 
-  /** Overloaded methods for unstructured logging **/
-  inline void log(nostd::string_view message) noexcept
+  /**
+   * Writes a log.
+   * @param severity The log's severity
+   * @param message The message to log
+   */
+  inline void Log(Severity severity, nostd::string_view message) noexcept
   {
-    // Set severity to the default then call log(Severity, String message) method
-    log(Severity::kDefault, message);
+    // Create a LogRecord to hold this information
+    LogRecord r;
+    r.severity = severity;
+    r.body     = message;
+
+    // Call the main Log(LogRecord) method
+    Log(r);
   }
 
-  inline void log(Severity severity, nostd::string_view message) noexcept
+  /**
+   * Writes a log.
+   * @param severity The log's severity
+   * @param name The name of the log
+   * @param message The message to log
+   */
+  inline void Log(Severity severity, nostd::string_view name, nostd::string_view message) noexcept
```

Comment:
Are there any plans to support specifying a format for the log message? Like this:
```c++
logger->Log(Severity::kError, "Log Name") << "Found " << num_cookies << " cookies";
```

Without a feature like this, the API might be pretty cumbersome to use.

Reply:
Yea, there definitely are plans for this ([based from this comment](https://github.com/open-telemetry/opentelemetry-cpp/issues/356#issuecomment-707128159)), but due to time constraints we only have time to PR an MVP logging API. I'll open up an issue to support formatting log messages this after this PR gets merged.